src/main/java/edu/cornell/kfs/sys/dataaccess/impl/AutoCancelBatchDaoJdbc.java

```diff
@@ -0,0 +1,206 @@
+package edu.cornell.kfs.sys.dataaccess.impl;
+
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+import edu.cornell.kfs.sys.batch.AutoCancelBatchStep;
+import edu.cornell.kfs.sys.dataaccess.AutoCancelBatchDao;
+import edu.cornell.kfs.sys.dataaccess.DbmsOutput;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.krad.document.Document;
+import org.kuali.kfs.krad.service.BusinessObjectService;
+import org.kuali.kfs.krad.service.DocumentService;
+import org.kuali.kfs.krad.service.SessionDocumentService;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.krad.workflow.service.WorkflowDocumentService;
+import org.kuali.rice.core.framework.persistence.jdbc.dao.PlatformAwareDaoBaseJdbc;
+import org.kuali.rice.kew.api.KewApiServiceLocator;
+import org.kuali.rice.kew.api.doctype.DocumentType;
+import org.kuali.rice.kew.api.document.DocumentStatus;
+import org.kuali.rice.kew.api.exception.WorkflowException;
+
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class AutoCancelBatchDaoJdbc extends PlatformAwareDaoBaseJdbc implements AutoCancelBatchDao {
+
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(AutoCancelBatchDaoJdbc.class);
+
+    private BusinessObjectService businessObjectService;
+    private DocumentService documentService;
+    private ParameterService parameterService;
+    private SessionDocumentService sessionDocumentService;
+    private WorkflowDocumentService workflowDocumentService;
+
+    private Collection<String> cancelDocumentTypes = Collections.emptySet();
+
+    /**
+     * @see AutoCancelBatchDao#cancelFYIsAndAcknowledgements()
+     */
+    @Override
+    public boolean cancelFYIsAndAcknowledgements() {
+        Connection conn = null;
+        CallableStatement cs = null;
+        DbmsOutput dbmsOutput = null;
+        boolean result;
+
+        try {
+            conn = getJdbcTemplate().getDataSource().getConnection();
+            cs = conn.prepareCall("{call CYNERGY.AUTO_CANCEL_FYI_ACK}");
+            dbmsOutput = new DbmsOutput(conn);
+            dbmsOutput.enable(1000000);
+            result = cs.execute();
+            dbmsOutput.show();
+        } catch (SQLException e) {
+            throw new RuntimeException("Issue encountered with cancelFYIsAndAcknowledgements", e);
+        } finally {
+            if (dbmsOutput != null) {
+                try {
+                    dbmsOutput.close();

```

Comment:
It looks like the current implementation of the DbmsOutput.close() method has the potential to leave some of its internal statements open if one of them throws an exception on close. If fixing that is out of scope for this PR, then we should at least create a new user story to get this fixed. (DbmsOutput is a CU-specific class, so we don't need to worry about fixing it at the KualiCo level.)